test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -8,43 +8,63 @@
 import java.util.Collection;
 import java.util.Iterator;
 
+import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
 
 import games.strategy.triplea.delegate.Matches;
 import games.strategy.util.IntegerMap;
 

```

Comment:
You're missing the `@RunWith(MockitoJunit4Runner.class)`
That is what will parse the class for the `@Mock` annotations. Without it the `@Mock` do nothing.